tests/PHPStan/Analyser/data/bug-4207.php

```diff
@@ -5,5 +5,6 @@
 use function PHPStan\Testing\assertType;
 
 function (): void {
-	assertType('array<int, int>&nonEmpty', range(1, 10000));
+	assertType('array<int, int<1, 10000>>&nonEmpty', range(1, 10000));
+	assertType('array<int, int>&nonEmpty', range(1, 20000, 2));
```

Comment:
since the new "optimized case" only supports step-length=1, we add a new test-case here with step-length=2 so the rather complex big-range logic is still covered